Lib/asyncio/proactor_events.py

```diff
@@ -160,22 +160,30 @@ def __init__(self, loop, sock, protocol, waiter=None,
 
     def pause_reading(self):
         if self._closing:
-            raise RuntimeError('Cannot pause_reading() when closing')
-        if self._paused:
-            raise RuntimeError('Already paused')
-        self._paused = True
-        if self._loop.get_debug():
-            logger.debug("%r pauses reading", self)
+            return
+        elif self._paused:
+            if self._loop.get_debug():
+                logger.debug(
+                    "%r: ignoring pause_reading() call; already paused",
+                    self)
+        else:
+            self._paused = True
+            if self._loop.get_debug():
+                logger.debug("%r pauses reading", self)
 
     def resume_reading(self):
-        if not self._paused:
-            raise RuntimeError('Not paused')
-        self._paused = False
         if self._closing:
             return
```

Comment:
I understand why you want to remove the exception from `pause_reading`. But rising an error on trying to `resume_reading` on a closed transport seems reasonable.